tests/src/Forms/ClientFormTest.php

```diff
@@ -4,16 +4,72 @@
 
 namespace SimpleSAML\Test\Module\oidc\Forms;
 
-use SimpleSAML\Module\oidc\Forms\ClientForm;
 use PHPUnit\Framework\TestCase;
+use SimpleSAML\Module\oidc\Forms\ClientForm;
+use SimpleSAML\Module\oidc\Forms\Controls\CsrfProtection;
+use SimpleSAML\Module\oidc\ModuleConfig;
 
 /**
  * @covers \SimpleSAML\Module\oidc\Forms\ClientForm
  */
 class ClientFormTest extends TestCase
 {
-    public function testIncomplete(): never
+    /** @var CsrfProtection */
+    protected CsrfProtection $csrfProtection;
+    /**
+     */
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $_SERVER['REQUEST_URI'] = '/';
+        $this->csrfProtection =  $this->getMockBuilder(CsrfProtection::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+    }
+
+    public static function validateOriginProvider(): array
     {
-        $this->markTestIncomplete();
+        return [
+            ['https://example.com', true],
+            ['https://example.com.', true],
+            ['http://foo.', true],
+            ['http://foo', true],
```

Comment:
@cicnavi What do you think about allowing hostnames that are not fully qualified? We have not encountered a developer setup using that, but I imagine it is possible.

Reply:
I also think it's fine to have non FQDN. And it can also be an IP address:  https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin

Reply:
I added a couple of more test cases that confirm that the regex works correctly for IP addresses.